src/coreclr/pgosupport.cmake

```diff
@@ -52,17 +52,17 @@ function(add_pgo TargetName)
                 add_compile_definitions(WITH_NATIVE_PGO)
             else(CLR_CMAKE_HOST_WIN32)
                 if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE OR UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)
-                    if((CMAKE_CXX_COMPILER_ID MATCHES "Clang") AND (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16))
+                    if((CMAKE_CXX_COMPILER_ID MATCHES "Clang") AND (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 20))
```

Comment:
nit: extract it in a property, something like `set(CLR_PGO_CLANG_VER 20)`

Reply:
yeah I'm looking at whether we can remove the need for manual versions completely

Reply:
straightforward way is to just write a file `clang -dumpversion > clang-version.txt` and add it to the optimization package. we already have a path, so we can read it in msbuild target for comparison.

Reply:
straightforward way is to just write a file `clang -dumpversion > clang-version.txt` on build machine  which creates the optimization package and include the version file. we already have a path in this context, so we can read it in cmake for comparison:

`file(READ "${CLR_CMAKE_OPTDATA_PATH}/clang-version.txt" PGO_BUILD_CLANG_VERSION)`

Reply:
I found an easier way, we can just pass the profile data to the compile time check we already do for lto, clang will complain if it can't read the profile.

Reply:
much better! :)